res/controllers/DJControl-Compact-scripts.js

```diff
@@ -0,0 +1,113 @@
+// Mode button does give out midi, but the selection itself is done inside the
+// controller -- no control over which mode is selected for instance.
+
+HercDJCompact = new function() {
+   this.group = "[Master]";
+}
+
+HercDJCompact.init = function(id) {
+    scratch = false;
+    scratch_timer = [];
+    scratch_timer_on = [];
+
+    engine.connectControl("[Recording]", "status",
+            "HercDJCompact.OnRecordingStatusChange");
+
+    // Tell controller to send midi to update knob and slider positions.
+    midi.sendShortMsg(0xB0, 0x7F, 0x7F);
+
+    // But the rate values get messed up, so reset them
+    engine.setValue("[Channel1]", "rate_set_default", 1.0);
+    engine.setValue("[Channel2]", "rate_set_default", 1.0);
+};
+
+HercDJCompact.shutdown = function() {
+    // toggle all lights off.
+    for (i = 0x01; i < 0x57; i++) {
+        midi.sendShortMsg(0x90, i, 0x00);
+    }
+};
+
+HercDJCompact.controls = {
+    // "name" is just for reference in the code.
+    "inputs": {
+        0x30: { "name": "jog", "channel": 1, "group": "[Channel1]"},
+        0x31: { "name": "jog", "channel": 2, "group": "[Channel2]"},
+        0x37: { "name": "pitch", "channel": 1, "group": "[Channel1]"},
+        0x38: { "name": "pitch", "channel": 2, "group": "[Channel2]"},
+    }
+};
+
+HercDJCompact.scratch = function (group, control, value, status) {
+    scratch = value > 0;
+};
+
+HercDJCompact.jog_wheel = function (group, control, value, status) {
+    var input = HercDJCompact.controls.inputs[control];
+    // If the high bit is 1, convert to a negative number
+    if (value & 0x40) {
+        value = value - 0x80;
+    }
+    if (scratch) {
+        if (value != 0) {
```

Comment:
Please use `!==` and `===` rather than `!=` and `==` to avoid unintended oddities with JS type conversions